src/main/java/com/commercetools/sync/commons/helpers/BaseSyncOptions.java

```diff
@@ -0,0 +1,129 @@
+package com.commercetools.sync.commons.helpers;
+
+import com.commercetools.sync.commons.CTPUtils;
+import com.commercetools.sync.services.TypeService;
+import com.commercetools.sync.services.impl.TypeServiceImpl;
+import io.sphere.sdk.client.BlockingSphereClient;
+import io.sphere.sdk.client.SphereClientConfig;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+public class BaseSyncOptions {
+    // CTP Client configuration
+    private SphereClientConfig clientConfig;
+    private long timeout = 30;
+    private TimeUnit timeoutTimeUnit = TimeUnit.SECONDS;
+    private BlockingSphereClient ctpClient;
+
+    // Custom Callbacks
+    private BiConsumer<String, Throwable> updateActionErrorCallBack;
+    private Consumer<String> updateActionWarningCallBack;
+
+    // Services
+    private TypeService typeService;
+
+    // Additional Options, still should be implemented GITHUB ISSUE#5
+
+    // Add additional localizations without deleting existing ones.
+    private boolean removeOtherLocales = true;
+
+    // Add additional Set entries without deleting existing ones.
+    private boolean removeOtherSetEntries = true;
+
+    //  Add additional collection (e.g. Assets, Images etc.) entries without deleting existing ones.
+    private boolean removeOtherCollectionEntries = true;
+
+    // Add additional object properties (e.g. custom fields, product attributes, etc..) without deleting existing ones.
+    private boolean removeOtherProperties = true;
+
+
+    public BaseSyncOptions(@Nonnull final String ctpProjectKey,
+                           @Nonnull final String ctpClientId,
+                           @Nonnull final String ctpClientSecret) {
+        this.clientConfig = SphereClientConfig.of(ctpProjectKey,
+                ctpClientId, ctpClientSecret);
+        ctpClient = CTPUtils.createClient(clientConfig, timeout, timeoutTimeUnit);
```

Comment:
As I commented above:
4d2f907

Regarding, the client it is also injected now in this commit.